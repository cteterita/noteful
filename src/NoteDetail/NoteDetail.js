import React from 'react';
import PropTypes from 'prop-types';
import './NoteDetail.css';
import Context from '../Context';

class NoteDetail extends React.Component {
  static contextType = Context;

  static defaultProps = {
    noteId: '',
  };

  static propTypes = {
    noteId: PropTypes.string,
    push: PropTypes.func.isRequired,
  }

  deleteNote = () => {
    const { noteId, push } = this.props;
    const { deleteNote } = this.context;
    deleteNote(noteId);
    push('/');
  }

  render() {
    const { noteId } = this.props;
    const { notes } = this.context;
    const note = notes.find((n) => n.id === noteId);
    if (note) {
      const { name, modified, content } = note;
      return (
        <div className="NoteDetail">
          <div className="NotePreview">
            {name}
            <br />
            {modified}
            <button
              className="delete-note-button"
              type="submit"
              onClick={this.deleteNote}
            >
              Delete
            </button>
          </div>
          {content}
        </div>
      );
    }
    return <></>;
  }
}

export default NoteDetail;
