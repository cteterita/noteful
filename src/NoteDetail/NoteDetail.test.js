import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import NoteDetail from './NoteDetail';
import Context from '../Context';

describe('NoteDetail component', () => {
  it('renders as expected', () => {
    const props = {
      match: {
        params: {
          noteId: '123',
        },
      },
      history: {
        push: jest.fn(),
      },
    };
    const context = {
      notes: [
        {
          name: 'note1',
          id: '123',
          content: 'Note content',
        },
      ],
      deleteNote: jest.fn(),
    };
    const wrapper = shallow((
      <Context.Provider notes={context.notes} deleteNote={context.deleteNote}>
        <NoteDetail match={props.match} history={props.history} />
      </Context.Provider>
    ));
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
