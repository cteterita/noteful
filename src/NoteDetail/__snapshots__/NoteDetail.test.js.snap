// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteDetail component renders as expected 1`] = `
<NoteDetail
  history={
    Object {
      "push": [MockFunction],
    }
  }
  match={
    Object {
      "params": Object {
        "noteId": "123",
      },
    }
  }
/>
`;
